src/main/java/practice/StreamPractice.java

```diff
@@ -59,22 +86,33 @@ public Double getOddNumsAverage(List<Integer> numbers) {
      * return the names of all cats whose owners are women from `femaleAge` years old inclusively.
      */
     public List<String> getCatsNames(List<Person> peopleList, int femaleAge) {
-        return Collections.emptyList();
+        Predicate<Person> personPredicate = person -> person.getSex() == Person.Sex.WOMAN
+                && person.getAge() >= femaleAge;
+        return peopleList.stream()
+                .filter(personPredicate)
+                .map(Person::getCats)
+                .flatMap(Collection::stream)
```

Comment:
```suggestion
                .flatMap(el -> el.getCats().stream())
```another line want to live))